pkg/storage/snapshot/snapshot.go

```diff
@@ -704,14 +704,30 @@ func (ctrl *VMSnapshotController) getSnapshotPVC(namespace, volumeName string) (
 	return nil, nil
 }
 
-func (ctrl *VMSnapshotController) getVolumeSnapshotClass(storageClassName string) (string, error) {
+func (ctrl *VMSnapshotController) getVolumeSnapshotClassName(storageClassName string) (string, error) {
 	obj, exists, err := ctrl.StorageClassInformer.GetStore().GetByKey(storageClassName)
 	if !exists || err != nil {
 		return "", err
 	}
 
 	storageClass := obj.(*storagev1.StorageClass).DeepCopy()
 
+	obj, exists, err = ctrl.StorageProfileInformer.GetStore().GetByKey(storageClassName)
+	if err != nil {
+		return "", err
+	}
+	if exists {
+		storageProfile := obj.(*cdiv1.StorageProfile).DeepCopy()
```

Comment:
Do you actually need a whole copy? wouldn't `obj.(*cdiv1.StorageProfile)` suffice?